lib/python/rose/config_editor/ops/group.py

```diff
@@ -52,6 +53,189 @@ def __init__(self, data, util, reporter, undo_stack, redo_stack,
         self.update_ns_sub_data_func = update_ns_sub_data_func
         self.reload_ns_tree_func = reload_ns_tree_func
 
+    def apply_diff(self, config_name, config_diff, origin_name=None,
+                   triggers_ok=False, is_reversed=False):
+        """Apply a rose.config.ConfigNodeDiff object to the config."""
+        state_reason_dict = {
+            rose.config.ConfigNode.STATE_NORMAL: {},
+            rose.config.ConfigNode.STATE_USER_IGNORED: {
+                rose.variable.IGNORED_BY_USER:
+                rose.config_editor.IGNORED_STATUS_MACRO
+            },
+            rose.config.ConfigNode.STATE_SYST_IGNORED: {
+                rose.variable.IGNORED_BY_SYSTEM:
+                rose.config_editor.IGNORED_STATUS_MACRO
+            }
+        }
+        nses = []
```

Comment:
It is namespaces (page addresses). Things tend to be organised and updated by namespace, although really only sections and options actually exist.